integration_tests/tests/events/data.rs

```diff
@@ -135,10 +135,11 @@ async fn transaction_execution_should_produce_events(
     executable: impl Into<Executable> + Send,
 ) -> Result<()> {
     let network = NetworkBuilder::new().start().await?;
-    let mut events_stream = network
+    let events_stream = network
         .client()
-        .listen_for_events_async([DataEventFilter::Any])
+        .listen_for_events([DataEventFilter::Any])
         .await?;
+    pin_mut!(events_stream);
```

Comment:
New implementation of async streams forces to use `pin_mut!` on the receiving side. It is a trade-off I've made:

- Cons: introduces a bit of verbosity for caller
- Pros:
  - minimal API
  - caller decides what to do with the stream (it may not necessarily want to pin it?)